docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+     - :ref:`LumpedModel <lumped_model_api>`
+     - Static lumped data over the response_units
+     - xarray.Dataset (compatible with hydromt.vector.GeoDataset)
+     - | :py:attr:`~LumpedModel.response_units`
+       | :py:func:`~LumpedModel.set_response_units`
+       | :py:func:`~LumpedModel.read_response_units`
+       | :py:func:`~LumpedModel.write_response_units`
+   * - mesh
+     - :ref:`MeshModel <mesh_model_api>`
+     - Static mesh (unstructured grid(s)) data
+     - xugrid.UgridDataset
+     - | :py:attr:`~MeshModel.mesh`
+       | :py:func:`~MeshModel.set_mesh`
+       | :py:func:`~MeshModel.get_mesh`
+       | :py:func:`~MeshModel.read_mesh`
+       | :py:func:`~MeshModel.write_mesh`
+
+Reading & Writing model components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Because the way that models consume data is so hyper specialised, the main part of writing a plugin is writing the
+Input/Output (IO) methods to prepare data exactly as your model expects them, in the correct format. To do this you'll
+be overwriting most ``read_<component>`` or ``write_<component>`` methods in your model class. That will ensure that the
+data comes in and goes out expecting exactly as you want.
+
+In the default HydroMT CORE read and write methods, xarray objects are written as netcdf files, ``geoms`` as GeoJSON files by
+default but any geopandas driver is allowed, and ``config`` eithey as TOML, YAML or INI file. So again, you could decide to
+completely overwrite the parent read and write methods or change the defaults.
+
+Eg if your geoms should be written in GeoPackage format rather than GeoJSON and in a geometry folder in your model:
+
+.. code-block:: python
+
+  def write_geoms(self):
+    """ Write model geometries to a GeoPackage file at <root>/geometry/<name>.gpkg """
+
+    super().write_geoms(
+      fn = "geometry/{name}.gpkg",
+      driver = 'GPKG'
+    )
+
+If you are writting your own read/write functions, some good pratice steps of the function:
+
+1. For writting, check first if the object you are writting is not empty, then you can pass. Example:
+
+.. code-block:: python
+
+  if len(self.geoms) == 0:
+    self.logger.debug("No geoms data found, skip writing.")
+    return
+
+2. For reading or writting, check that you have the right permission. For example, in build mode ('w' or 'w+'),
+the model should be empty, so you should not be able to read existing model files. In some cases, python
+users may only want to read a model using HydroMT to visualise or plot data. As a protection, the model
+can then be opened in read only mode ('r') and HydroMT should not be able to overwrite any data. HydroMT
+`Model` class has two porperties you can use to check for this:
+
+- `self._assert_read_mode` if a model can be read
+- `self._assert_write_mode` if a model can be written
+3. If needed transform, and write the component to the right format and file(s). Don't hesitate to put
+some logger information for your user.
+
+Set model components
```

Comment:
In this section the main message should be to **not** change the set methods and properties which are defined in the Model class. So also showing the code is not necessary I think. It is however good to explain that the data is read upon first call of the property